src/MCPClient/lib/clientScripts/upload-archivesspace.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/python
+
+import argparse
+import logging
+import os
+from urlparse import urljoin
+
+from main.models import ArchivesSpaceDIPObjectResourcePairing
+
+# archivematicaCommon
+from archivesspace.client import ArchivesSpaceClient
+from elasticSearchFunctions import getDashboardUUID
+from xml2obj import mets_file
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+logger.addHandler(logging.FileHandler('/tmp/as_upload.log', mode='a'))
+
+
+def recursive_file_gen(mydir):
+    for root, dirs, files in os.walk(mydir):
+        for file in files:
+            yield os.path.join(root, file)
+
+
+def get_files_from_dip(dip_location, dip_name, dip_uuid):
+    # need to find files in objects dir of dip:
+    # go to dipLocation/dipName/objects
+    # get a directory listing
+    # for each item, set fileName and go
+    try:
+        mydir = dip_location + "objects/"
+        mylist = list(recursive_file_gen(mydir))
+
+        if len(mylist) > 0:
+            return mylist
+        else:
+            logger.error("no files in " + mydir)
+            raise ValueError("cannot find dip")
+            exit(2)
+    except Exception:
+        raise
+        exit(3)
+
+
+def get_uri(prefix, dip_uuid):
+    return urljoin(prefix, dip_uuid) + "/objects/"
+
+
+def get_pairs(dip_uuid):
+    return {pair.fileuuid: pair.resourceid for pair in ArchivesSpaceDIPObjectResourcePairing.objects.filter(dipuuid=dip_uuid)}
+
+
+def delete_pairs(dip_uuid):
+    ArchivesSpaceDIPObjectResourcePairing.objects.filter(dipuuid=dip_uuid).delete()
+
+
+def upload_to_archivesspace(files, client, xlink_show, xlink_actuate, object_type, use_statement, uri_prefix, dip_uuid, access_conditions, use_conditions, restrictions, dip_location):
+
```

Comment:
The other script still exists; I used it as a template, since this does mostly the same things with some customizations. Eventually might want to refactor into one script with two backends but that wasn't feasible at this point in time.